vaadin-testbench-shared/src/main/java/com/vaadin/testbench/ElementQuery.java

```diff
@@ -48,33 +57,153 @@ public class ElementQuery<T extends TestBenchElement> {
-        private final String operator;
-        private final String value;
-        private final boolean negate;
 
-        public AttributeMatch(String name, String operator, String value) {
-            this.name = name;
+        public enum Operator {
+            /**
+             * Attribute exists (with or without a value).
+             */
+            EXISTS(""),
+            /**
+             * Attribute value and given value match exactly.
+             */
+            MATCHES_EXACTLY("="),
+            /**
+             * Attribute value contains the given value.
+             */
+            CONTAINS("*="),
+            /**
+             * Attribute value contains a space-separated token
+             * that matches the given value.
+             */
+            CONTAINS_TOKEN("~="),
+            /**
+             * Attribute value begins with a space/hyphen-separated prefix
+             * that matches the given value.
+             * The prefix must be either the entire attribute value or
+             * the leading hyphen-separated segments of the attribute value.
+             */
+            CONTAINS_PREFIX("|="),
+            /**
+             * Attribute value begins with the given value.
+             */
+            BEGINS_WITH("^="),
+            /**
+             * Attribute value ends with the given value.
+             */
+            ENDS_WITH("$="),
+            /**
+             * Attribute does not exist (with or without a value).
+             */
+            NOT_EXISTS(EXISTS.op, true),
+            /**
+             * Attribute value and given value do not match exactly.
+             */
+            NOT_MATCHES_EXACTLY(MATCHES_EXACTLY.op, true),
+            /**
+             * Attribute value does not contain the given value.
+             */
+            NOT_CONTAINS(CONTAINS.op, true),
+            /**
+             * Attribute value must not contain a space-separated token
+             * that is prefixed with the given value.
+             * @see #CONTAINS_TOKEN
+             */
+            NOT_CONTAINS_TOKEN(CONTAINS_TOKEN.op, true),
+            /**
+             * Attribute value does not begin with a space/hyphen-separated prefix
+             * that matches the given value.
+             */
+            NOT_CONTAINS_PREFIX(CONTAINS_PREFIX.op, true),
+            /**
+             * Attribute value does not begin with the given value.
+             */
+            NOT_BEGINS_WITH(BEGINS_WITH.op, true),
+            /**
+             * Attribute value does not end with the given value.
+             */
+            NOT_ENDS_WITH(ENDS_WITH.op, true);
+
+            private final String op;
+            private final boolean neg;
+
+            Operator(String op, boolean neg) {
+                this.op = op;
+                this.neg = neg;
+            }
+
+            Operator(String op) {
+                this(op, false);
+            }
+
+            public String getOp() {
```

Comment:
As this becomes a public API, we should have a more friendly name, e.g. `getOperator()` :grin: 
Also needs Javadoc.
The same goes for `isNeg()`

Reply:
Agreed. So there would be no confusion by name reuse, I changed the enum's name to Comparison and spelled out the member/getter names.